spec/03-plugins/23-rate-limiting/04-access_spec.lua

```diff
@@ -12,6 +12,17 @@ local fmt = string.format
 local proxy_client = helpers.proxy_client
 
 
+local function it_with_retry(desc, test)
```

Comment:
This needs a big comment explaining why this is done and why it works, because adding retries is **NOT** a pattern that we should adopt in general for dealing with flaky tests:

```suggestion
-- This performs the test up to two times (and no more than two).
-- We are **not** retrying to "give it another shot" in case of a flaky test.
-- The reason why we allow for a single retry in this test suite is because
-- tests are dependent on the value of the current minute. If the minute
-- flips during the test (i.e. going from 03:43:59 to 03:44:00), the result
-- will fail. Since each test takes less than a minute to run, running it
-- a second time right after that failure ensures that another flip will
-- not occur. If the second execution failed as well, this mean that there
-- was an actual problem with the test.
local function it_with_retry(desc, test)
```